### PR TITLE
Automate the tests to create an LR claim using an API call

### PR DESCRIPTION
**Pull request by nishm**
### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-5004

### Change description ###
This PR will add create a claim (LR vs LR) via API

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: e2e/config.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    url: 'http://localhost:5555',
+  },
+  url: {
+    manageCase: process.env.URL || 'https://manage-case.demo.platform.hmcts.net/',
+    authProviderApi: process.env.SERVICE_AUTH_PROVIDER_API_BASE_URL || 'http://rpe-service-auth-provider-demo.service.core-compute-demo.internal',
+    ccdDataStore: process.env.CCD_DATA_STORE_URL || 'http://ccd-data-store-api-demo.service.core-compute-demo.internal',
+    dmStore:process.env.DM_STORE_URL || 'http://dm-store-demo.service.core-compute-demo.internal',
+    idamApi: process.env.IDAM_API_URL || 'https://idam-api.demo.platform.hmcts.net',
+    civilService: process.env.CIVIL_SERVICE_URL || 'http://civil-service-demo.service.core-compute-demo.internal',
+    // manageCase: process.env.URL || 'http://localhost:3333',
```
Comment (rajakm) [inline e2e/config.js:15]: Revert it to local

File: e2e/crossbrowser/supportedBrowsers.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+const LATEST_WINDOWS = 'Windows 11';
```
Comment (rajakm) [inline e2e/crossbrowser/supportedBrowsers.js:1]: We can remove this file as it is not needed for api test

File: e2e/fixtures/ccd/events.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+module.exports = {
```
Comment (rajakm) [inline e2e/fixtures/ccd/events.js:1]: We dont need the events.js file for Create Spec claim

File: e2e/fixtures/ccd/expectedEvents.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+const events = require('./events.js');
+
+module.exports = {
+  applicant_solicitor: {
+    CASE_ISSUED: [
+      events.NOTIFY_DEFENDANT_OF_CLAIM,
+      events.ADD_OR_AMEND_CLAIM_DOCUMENTS,
+    ],
```
Comment (rajakm) [inline e2e/fixtures/ccd/expectedEvents.js:8]: In this file, just applicant_solicitor Case_issued event is enough

File: e2e/fixtures/events/createClaimSpec.js [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+      },
+    };
+
+    switch (mpScenario) {
+      case 'ONE_V_ONE':
+        userData.userInput = {
+          ...userData.userInput,
+        };
+        break;
+      case 'ONE_V_TWO_TWO_LEGAL_REP':
```
Comment (rajakm) [inline e2e/fixtures/events/createClaimSpec.js:213]: just ONE_V_ONE case is enough, we can remove the other types

File: e2e/config.js (hunk 1 above)
Comment (nishm) [reply e2e/config.js:15]: Done

File: e2e/fixtures/ccd/expectedEvents.js (hunk 4 above)
Comment (nishm) [reply e2e/fixtures/ccd/expectedEvents.js:8]: Done

File: e2e/fixtures/ccd/events.js (hunk 3 above)
Comment (nishm) [reply e2e/fixtures/ccd/events.js:1]: Done

File: e2e/fixtures/events/createClaimSpec.js (hunk 5 above)
Comment (nishm) [reply e2e/fixtures/events/createClaimSpec.js:213]: Done

File: e2e/api/steps.js [hunk 6]
```diff
@@ -0,0 +1,472 @@
...
+
+    await assignSpecCase(caseId, multipartyScenario);
+    await waitForFinishedBusinessProcess(caseId);
+
+    //field is deleted in about to submit callback
+    deleteCaseFields('applicantSolicitor1CheckEmail');
+    return caseId;
+  },
+
+  notifyClaimDetails: async (user, caseId) => {
```
Comment (mounikahmcts) [inline e2e/api/steps.js:111]: notifyClaimDetails is not needed for CUI

File: e2e/helpers/accessibility/runner.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
+const HTMLCS = require('html_codesniffer');
```
Comment (mounikahmcts) [inline e2e/helpers/accessibility/runner.js:1]: this file is not required

File: e2e/helpers/browser_helper.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
+const testConfig = require('../config.js');
```
Comment (mounikahmcts) [inline e2e/helpers/browser_helper.js:1]: this file is not required

**Review by mounikahmcts — APPROVED** [review]

File: e2e/api/steps.js [hunk 9]
```diff
@@ -0,0 +1,472 @@
...
+  },
+};
+
+let caseId, eventName;
+let caseData = {};
+let mpScenario = 'ONE_V_ONE';
+
+module.exports = {
+
+  createUnspecifiedClaim: async (user, multipartyScenario, claimantType) => {
```
[the remark this answers is not in the document]
Comment (nishm) [reply e2e/api/steps.js]: Done

File: e2e/api/steps.js (hunk 6 above)
Comment (nishm) [reply e2e/api/steps.js]: Done

File: e2e/tests/ui_tests/smoke_test_civil.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+const mpScenario = 'ONE_V_ONE';
+
+let parentCaseNumber, childCaseNumber;
+
+Feature('CUI Ex Smoke tests');
+
+Scenario.skip('CUI Ex Tests', async ({I, api}) => {
+  parentCaseNumber = await api.createUnspecifiedClaim(
+    config.applicantSolicitorUser, mpScenario, 'Company');
+  await api.notifyClaim(config.applicantSolicitorUser, mpScenario, parentCaseNumber);
```
[the remark this answers is not in the document]
Comment (nishm) [reply e2e/tests/ui_tests/smoke_test_civil.js]: Done

File: e2e/tests/ui_tests/smoke_test_civil.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable no-unused-vars */
```
[the remark this answers is not in the document]
Comment (nishm) [reply e2e/tests/ui_tests/smoke_test_civil.js]: Done

File: e2e/helpers/browser_helper.js (hunk 8 above)
Comment (nishm) [reply e2e/helpers/browser_helper.js]: Done

File: e2e/helpers/accessibility/runner.js (hunk 7 above)
Comment (nishm) [reply e2e/helpers/accessibility/runner.js]: Done

File: e2e/config.js [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  adminUser: {
+    password: defaultPassword,
+    email: 'civil-admin@mailnesia.com',
+    type: 'admin',
+  },
+  definition: {
+    jurisdiction: 'CIVIL',
+    caseType: 'CIVIL',
+    caseTypeGA: 'GENERALAPPLICATION',
```
Comment (mounikahmcts) [inline e2e/config.js:49]: need to remove GA

File: e2e/config.js [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+  },
+  TestOutputDir: process.env.E2E_OUTPUT_DIR || 'test-results/functional',
+  TestForAccessibility: process.env.TESTS_FOR_ACCESSIBILITY === 'true',
+  runningEnv: process.env.ENVIRONMENT,
+  claimantSolicitorOrgId: process.env.ENVIRONMENT == 'demo' ? 'B04IXE4' : 'Q1KOKP2',
+  defendant1SolicitorOrgId: process.env.ENVIRONMENT == 'demo' ? 'DAWY9LJ' : '79ZRSOU',
```
Comment (mounikahmcts) [inline e2e/config.js:55]: remove defendant1SolicitorOrgId and defendant2SolicitorOrgId

File: e2e/fixtures/events/createClaimSpec.js [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+          specApplicantCorrespondenceAddressRequired: 'No',
+        },
+        Defendant: {
+          respondent1: respondent1WithPartyName,
+        },
+        LegalRepresentation: {
+          specRespondent1Represented: 'Yes',
+        },
+
+        DefendantSolicitorOrganisation: {
```
Comment (mounikahmcts) [inline e2e/fixtures/events/createClaimSpec.js:87]: not correct

File: e2e/fixtures/events/createClaimSpec.js [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+          },
+        },
+        specCorrespondenceAddress: {
+          specApplicantCorrespondenceAddressRequired: 'No',
+        },
+        Defendant: {
+          respondent1: respondent1WithPartyName,
+        },
+        LegalRepresentation: {
+          specRespondent1Represented: 'Yes',
```
Comment (mounikahmcts) [inline e2e/fixtures/events/createClaimSpec.js:84]: spec represented is No

File: e2e/reporters/accessibility-reporter/customReporter.js [hunk 16]
```diff
@@ -0,0 +1,59 @@
+const fs = require('fs');
```
Comment (mounikahmcts) [inline e2e/reporters/accessibility-reporter/customReporter.js:1]: remove this file

**Pull request closed without merging**